--- a/ultramodern/src/threadqueue.cpp
+++ b/ultramodern/src/threadqueue.cpp
@@ -15,7 +15,7 @@
     PTR(OSThread)* cur = queue_to_ptr(PASS_RDRAM queue_);
     OSThread* toadd = TO_PTR(OSThread, toadd_); 
     debug_printf("[Thread Queue] Inserting thread %d into queue 0x%08X\n", toadd->id, (uintptr_t)queue_);
-    while (*cur && TO_PTR(OSThread, *cur)->priority > toadd->priority) {
+    while (*cur && TO_PTR(OSThread, *cur)->priority >= toadd->priority) {
         cur = &TO_PTR(OSThread, *cur)->next;
     }
     toadd->next = (*cur);
@@ -43,14 +43,14 @@
 bool ultramodern::thread_queue_remove(RDRAM_ARG PTR(PTR(OSThread)) queue_, PTR(OSThread) t_) {
     debug_printf("[Thread Queue] Removing thread %d from queue 0x%08X\n", TO_PTR(OSThread, t_)->id, (uintptr_t)queue_);
 
-    PTR(PTR(OSThread)) cur = queue_;
-    while (cur != NULLPTR) {
-        PTR(OSThread)* cur_ptr = queue_to_ptr(PASS_RDRAM queue_);
-        if (*cur_ptr == t_) {
-            *cur_ptr = TO_PTR(OSThread, *cur_ptr)->next;
+    if (queue_ == NULLPTR) return false;
+    PTR(OSThread)* cur = queue_to_ptr(PASS_RDRAM queue_);
+    while (*cur != NULLPTR) {
+        if (*cur == t_) {
+            *cur = TO_PTR(OSThread, *cur)->next;
             return true;
         }
-        cur = TO_PTR(OSThread, *cur_ptr)->next;
+        cur = &TO_PTR(OSThread, *cur)->next;
     }
 
     return false;
